primitives/src/must_use.rs

```diff
@@ -0,0 +1,67 @@
+//! Contains wrapper type to annotate types with `must_use` attribute
+
+use core::borrow::{Borrow, BorrowMut};
+
+use derive_more::{AsMut, AsRef, Constructor, Deref, Display};
+
+/// Wrapper type to annotate types with `must_use` attribute
+///
+/// # Example
+/// ```
+/// use iroha_primitives::must_use::MustUse;
+///
+/// fn is_odd(num: i32) -> Result<MustUse<bool>, String> {
+///     if num < 0 {
+///         return Err(String::from("Number must be positive"));
+///     }
+///
+///     if num % 2 == 0 {
+///         Ok(MustUse::new(true))
+///     } else {
+///         Ok(MustUse::new(false))
+///     }
+/// }
+///
+/// if *is_odd(2).unwrap() {
+///     println!("2 is odd");
+/// }
+///
+/// // Will produce a warning, case `#[warn(unused_must_use)]` on by default
+/// // is_odd(3).unwrap();
+/// ```
+#[derive(
+    Constructor,
+    Debug,
+    Display,
+    Copy,
+    Clone,
+    PartialEq,
+    Eq,
+    PartialOrd,
+    Ord,
+    Hash,
+    AsRef,
+    AsMut,
+    Deref,
+)]
+#[must_use]
+pub struct MustUse<T>(T);
```

Comment:
```suggestion
pub struct MustUse<T>(pub T);
```

This will allow us to use `MustUse` in pattern matchin and destructuring.
I see no downsides in making field public.

Reply:
Ok, let's do it